blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java

```diff
@@ -209,7 +209,11 @@ public void onClick(View v) {
                 }
             }
         };
-        return buildIconFieldView(listener, com.google.blockly.android.R.drawable.mutator_icon_24dp);
+        if (!getWorkspaceHelper().getWorkspaceView().getController().getWorkspace().isRootBlock(block.getRootBlock())) {
```

Comment:
Isn't
```!getWorkspaceHelper().getWorkspaceView().getController()```
the same as `this.mController` of the `BlockViewFactory`?
This means the new `getController()` added to `WorkspaceView` is not necessary.

Reply:
Neither ```BlockViewFactory``` nor ```VerticalBlockViewFactory``` seem to have a ```this.mController``` property.

Reply:
Whoops.  Misread mContext.  But `block.getController()` does exist:
https://github.com/google/blockly-android/blob/develop/blocklylib-core/src/main/java/com/google/blockly/model/Block.java#L197

Reply:
Done!